apps/web/src/actions/copilot/requestSuggestion.ts

```diff
@@ -0,0 +1,61 @@
+'use server'
+
+import { BadRequestError } from '@latitude-data/core/lib/errors'
+import { DocumentVersionsRepository } from '@latitude-data/core/repositories'
+import { env } from '@latitude-data/env'
+import { createSdk } from '$/app/(private)/_lib/createSdk'
+import { z } from 'zod'
+
+import { authProcedure } from '../procedures'
+
+export const requestSuggestionAction = authProcedure
+  .createServerAction()
+  .input(
+    z.object({
+      projectId: z.number(),
+      commitUuid: z.string(),
+      documentUuid: z.string(),
+      request: z.string(),
+    }),
+  )
+  .handler(async ({ ctx, input }) => {
+    if (!env.DATASET_GENERATOR_WORKSPACE_APIKEY) {
+      throw new BadRequestError('DATASET_GENERATOR_WORKSPACE_APIKEY is not set')
+    }
+    if (!env.COPILOT_PROJECT_ID) {
+      throw new BadRequestError('COPILOT_PROJECT_ID is not set')
+    }
+    if (!env.COPILOT_CODE_SUGGESTION_PROMPT_PATH) {
+      throw new BadRequestError(
+        'COPILOT_CODE_SUGGESTION_PROMPT_PATH is not set',
+      )
+    }
```

Comment:
All the time. At some point we have to unify this mambo jambo

Reply:
agree